app/controllers/session/sso_controller.rb

```diff
@@ -8,20 +8,28 @@
 class Session::SsoController < SessionController
 
   before_action :keycloak_cookie
+  skip_before_action :redirect_if_no_private_key
 
   def create
     cookies.permanent[:keycloak_token] = user_authenticator.token(params) if params[:code].present?
-    unless user_authenticator.authenticate!
-      return redirect_to user_authenticator.keycloak_login
+    return redirect_to user_authenticator.keycloak_login unless user_authenticator.authenticate!
+
+    unless create_session(keycloak_client.user_pk_secret)
+      return redirect_if_decryption_error
```

Comment:
Mixing guard clause and normal unless blocks? Because of rubocop? :upside_down_face: